src/main/java/com/cs525/reversi/models/Algorithm.java

```diff
@@ -0,0 +1,8 @@
+package com.cs525.reversi.models;
+
+import java.util.List;
+
+public interface Algorithm {
+
+	 Move decideMove(List<MoveScore> movePoints , List<MatrixRow> gameBoard);
```

Comment:
@MustafaMosad I think a better return type would be the `MoveScore` since that would also have the cell locations that would be flipped as a result of that move.... it would help to have this data when saving that move because now we can easy persist the list of cell locations to flip in the db.